src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java

```diff
@@ -373,6 +373,10 @@ public Object deserializeFromObject(JsonParser p, DeserializationContext ctxt) t
                 _handleTypedObjectId(p, ctxt, bean, id);
             }
         }
+        // [databind#3838]: since 2.16 Uniform handling of missing objectId
+        if (_objectIdReader != null && p.getCurrentToken() == JsonToken.END_OBJECT) {
```

Comment:
Ahhhhhh. It is only for the specific case of empty JSON Object. That's the piece I was missing, wondering how could this be safe. I'll update comment and

Reply:
> It is only for the specific case of empty JSON Object

Looking back, making a comment about the empty part would have made things clearer 😄 I will add updating comment (or checking that comment was made)  to my todo list also 👍🏻👍🏻